projects/components/src/viewer/code-viewer/code-viewer.component.ts

```diff
@@ -145,14 +159,66 @@ export class CodeViewerComponent implements AfterViewInit, OnChanges, OnDestroy
     this.domMutationObserver.disconnect();
   }
 
+  public get isCodeEmpty(): boolean {
+    return isEmpty(this.code);
+  }
+
   public isLineHighlighted(lineNum: number): boolean {
     return (
       !isEmpty(this.highlightText) && this.codeLines[lineNum].toLowerCase().includes(this.highlightText.toLowerCase())
     );
   }
 
-  public onSearch(searchText: string): void {
-    this.searchText = searchText;
+  public onSearch(searchString: string): void {
+    // For case-insensitive search
+    const searchText = searchString.toLowerCase();
+    const codeTexts = this.codeTexts.map(text => text.toLowerCase());
+
+    // Remove existing child background elements
+    this.codeLineBackgroundElements.forEach(codeLineBackgroundElement => {
+      const codeLineBackgroundElemNode: Node = codeLineBackgroundElement.nativeElement;
+
+      while (codeLineBackgroundElemNode.firstChild) {
+        codeLineBackgroundElemNode.removeChild(codeLineBackgroundElemNode.firstChild);
+      }
+    });
```

Comment:
instead of this, see if you can use the `replaceChildren()` method to just clear everything in one go.

Reply:
Used innerHTMl method

Reply:
that works too. `replaceChildren()` is a fairly new introduction https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceChildren